GitUI/UserControls/RevisionGrid/RevisionDataGridView.cs

```diff
@@ -425,13 +425,19 @@ public void LoadingCompleted()
                 _loadedToBeSelectedRevisionsCount = ToBeSelectedObjectIds.Count;
             }
 
-            if (_loadedToBeSelectedRevisionsCount > 0 && _revisionGraph.Count > 0)
+            if (_revisionGraph.Count == 0)
```

Comment:
If there is nothing to select, the loop is not needed.
Or did you have any other reason?

```suggestion
            if (_revisionGraph.Count == 0 || _loadedToBeSelectedRevisionsCount == 0)
```

Reply:
The issue is caused by not setting `RowCount` in some cases. That's why this condition should be independent of all these selection things.
`RowCount` can only be set after switching to the main thread.

> If there is nothing to select, the loop is not needed.

Me too thought that the while loop below might have become superfluous after calling `SetRowCountAndSelectRowsIfReady()`. Though I was not sure and did not want to dig deeper.

Reply:
> RowCount can only be set after switching to the main thread.

Then set RowCount early in the async code and exit before the while if no revisions at all or none to select?

Reply:
I'd say that's what is done. What have I missed?
```
            if (_revisionGraph.Count == 0)
            {
                MarkAsDataLoadingComplete();
            }
            else
            {
                // Rows have not been selected yet
                ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
                {
                    await this.SwitchToMainThreadAsync();

                    SetRowCountAndSelectRowsIfReady();

                    if (_toBeSelectedGraphIndexesCache.Value.Count == 0)
                    {
                        // Nothing to select or interrupted
                        MarkAsDataLoadingComplete();
                        return;
                    }
```

Reply:
Yes, it seem to work
The loading is complex, I need to look really close at it even if I have been involved in this part.